server/handlers.go

```diff
@@ -716,6 +721,27 @@ func (s *Server) handleAuthCode(w http.ResponseWriter, r *http.Request, client s
 		return
 	}
 
+	// Check for code challenge validity
+	if authCode.CodeChallenge != "" {
+		if verifier == "" {
+			s.tokenErrHelper(w, errAccessDenied, "Require code_verifier", http.StatusUnauthorized)
+			return
+		}
+		var challenge string
+		switch strings.ToLower(authCode.CodeChallengeMethod) {
+		case "s256":
```

Comment:
Could we have constants? something like [this](https://github.com/dexidp/dex/blob/master/server/oauth2.go#L124)